Lib/test/test_unpack.py

```diff
@@ -167,5 +220,52 @@ def test_extended_oparg_not_ignored(self):
             y = unpack_400(range(400))
             self.assertEqual(y, 399)
 
+    def test_exception_context_when_len_fails(self):
+        """When `__len__()` raises an Exception, preserve exception context"""
+        code = textwrap.dedent(
+            f"""
+            class CustomSeq:
+                def __len__(self):
+                    raise RuntimeError
+
+                def __getitem__(self, i):
+                    return i*2
+
+            x, y, z = CustomSeq()
+            """
+        )
+        with self.assertRaises(ValueError) as cm:
+            exec(code)
+
+        # Force processing by traceback module
+        traceback.format_exception(cm.exception)
```

Comment:
AFAIK, `traceback` is not guaranteed to modify the exception; the traceback is what matters. I'd go for something like:
```suggestion
        text = ''.join(traceback.format_exception(cm.exception))
```
and then look for `RuntimeError` and the message in `text`.

Or maybe use a doctest for this case too.

Reply:
Unfortunately `doctest` doesn't handle `BaseExceptions`.